core/src/main/scala/chisel3/probe/Probe.scala

```diff
@@ -40,8 +46,12 @@ object Probe extends ProbeBase with SourceInfoDoc {
     */
   def apply[T <: Data](source: => T): T = macro chisel3.internal.sourceinfo.ProbeTransform.sourceApply[T]
 
+  def apply[T <: Data](source: => T, color: layer.Layer): T =
+    macro chisel3.internal.sourceinfo.ProbeTransform.sourceApplyWithColor[T]
+
   /** @group SourceInfoTransformMacro */
-  def do_apply[T <: Data](source: => T)(implicit sourceInfo: SourceInfo): T = super.apply(source, false)
+  def do_apply[T <: Data](source: => T, color: Option[layer.Layer] = None)(implicit sourceInfo: SourceInfo): T =
+    super.apply(source, false, color)
```

Comment:
```suggestion
  def do_apply[T <: Data](source: => T)(implicit sourceInfo: SourceInfo): T = do_apply(source, None)

  /** @group SourceInfoTransformMacro */
  def do_apply[T <: Data](source: => T, color: Option[layer.Layer])(implicit sourceInfo: SourceInfo): T =
    super.apply(source, false, color)
```
This can be made to preserve binary compatibility by using overloading instead of default arguments

Reply:
Thanks! Fixed in https://github.com/chipsalliance/chisel/pull/3744/commits/87181a9cb4651f70b9e69812fccb3d859efcfadb.